### PR TITLE
Add job index number to jobs.json

### LINKED ISSUE #918: Add job index number to jobs.json
For debugging purposes, the job index would be nice to have inside each entry in `jobs.json`, that matches with the index suffix in the stdout/stderr files.

### PR DESCRIPTION
**Pull request by hnformentin**
**Issue**
Resolves #918

### CODE REVIEW

File: lib/job_queue/ext_job.cpp [hunk 1]
```diff
@@ -866,17 +866,24 @@ static void __fprintf_python_arg_types(FILE * stream,
...
 
-void ext_job_json_fprintf(const ext_job_type * ext_job, FILE * stream, const subst_list_type * global_args) {
+void ext_job_json_fprintf(const ext_job_type * ext_job, int job_index, FILE * stream, const subst_list_type * global_args) {
   const char * null_value = "null";
+  
+  char * file_stdout_index = NULL;
+  char * file_stderr_index = NULL;
+  
+  file_stdout_index = util_alloc_sprintf("%s.%d",ext_job->stdout_file, job_index);
+  file_stderr_index = util_alloc_sprintf("%s.%d",ext_job->stderr_file, job_index);
```
Comment (sondreso) [inline lib/job_queue/ext_job.cpp:876]: This is leaking memory, you also need to add 
```
free( file_stdout_index );
free( file_stderr_index );
```
after you are done using the strings.

File: python/tests/job_runner/test_jobmanager.py [hunk 2]
```diff
@@ -153,13 +153,13 @@ def test_run_output_rename(self):
...
-        for d in dir_list:
+        for job_index in dir_list:
             job = {
                 "name": "MKDIR",
                 "executable": "/bin/mkdir",
-                "stdout": "mkdir_out",
-                "stderr": "mkdir_err",
-                "argList": ["-p", "-v", d],
+                "stdout": "mkdir_out.%s" % job_index,
+                "stderr": "mkdir_err.%s" % job_index,
```
Comment (sondreso) [inline python/tests/job_runner/test_jobmanager.py]: Would prefer to use the `format` function instead

File: python/tests/job_runner/test_jobmanager.py [hunk 3]
```diff
@@ -171,11 +171,11 @@ def test_run_multiple_ok(self):
...
-        for index, dir_number in enumerate(dir_list):
-            self.assertTrue(os.path.isdir(dir_list[index]))
-            self.assertTrue(os.path.isfile("mkdir_out.%d" % index))
-            self.assertTrue(os.path.isfile("mkdir_err.%d" % index))
-            self.assertEqual(0, os.path.getsize("mkdir_err.%d" % index))
+        for dir_number in dir_list:
+            self.assertTrue(os.path.isdir(dir_number))
+            self.assertTrue(os.path.isfile("mkdir_out.%s" % dir_number))
+            self.assertTrue(os.path.isfile("mkdir_err.%s" % dir_number))
+            self.assertEqual(0, os.path.getsize("mkdir_err.%s" % dir_number))
```
Comment (sondreso) [inline python/tests/job_runner/test_jobmanager.py]: Would also prefer to use the `format` function here if we are touching the code anyways!

**Comment by sondreso** [comment]
Except for the small comments above i think this looks good! Will approve after this is fixed :+1:

**Comment by hnformentin** [comment]
Test this please!

**Comment by hnformentin** [comment]
Test this please!

**Comment by hnformentin** [comment]
Test this please!

**Comment by hnformentin** [comment]
Test this please!

**Review by sondreso — APPROVED** [review]
LGTM! Nice work!

**Pull request merged**